### PR TITLE
Feature/Calendar Events Pane

### PR DESCRIPTION
**Pull request by Barde-E**
### Summary

Implement an events pane to the calendar view and reorganized data files

### What's new

- In the `CalendarGridContent` composable, the new component `EventsPane` is called to show events on the screen
- This new `EventsPane` composable render `EventBlock` for all events in the current days
- Every Event is represented on the screen by a `EventBlock` composable
- A new `EventPositionUtil` have been created to help handling the placement of the event
- In general and among the composable, two new parameters are used : `dateRange` and `events`. They are both used to render correctly the events
- Mock data have been created to simulate events to show on the screen
- Some files representing data (`Days.kt`, `LocalDateRange` and `TimeSpan`) have been moved from the `utils` folder to a new `data` folder

### Follow-up
- Implement the `DayHeaderRow` composable in the `CalendarGridContent` to display the days and dates

This PR is related to the task #89
Depends on previous parts : #96 empty screen scaffold, #98 grid background and #99 time axis

### CODE REVIEW

**Review by noa-flo — CHANGES_REQUESTED** [review]
Looks great overall — solid structure and clean integration of the events pane 👍
I'd just suggest making the KDoc a bit more Java-like (use @param, @return, @throws, and imperative phrasing).

Also, improving test coverage for overlapping and edge-case events so that SonarCloud has no issue (actually one pending).
 
TODO comments should not be present when merge.

**Review by HB616 — COMMENTED** [review]
Overall:
This is a very clean and well-structured implementation of the calendar event layout. The composables are modular, well-documented, and clearly designed with future extensibility in mind (e.g., overlap handling and orientation changes).

File: app/src/main/java/com/android/sample/ui/calendar/components/EventBlock.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                text = event.assigneeName,
+                color = textColor.copy(alpha = 0.8f),
+                fontSize = 10.sp,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+          }
+
+          // Time information
+          val formatter = DateTimeFormatter.ofPattern("HH:mm", Locale.getDefault())
```
[the remark this answers is not in the document]
Comment (noa-flo) [reply app/src/main/java/com/android/sample/ui/calendar/components/EventBlock.kt]: I agree, this should be done with remember.

**Review by noa-flo — APPROVED** [review]
All issues fixed and 98.1% of line coverage, I approve this PR.
